app/helpers/notifications_helper.rb

```diff
@@ -232,7 +232,11 @@ def repo_filter_link(param, repo_name, count)
   end
 
   def sidebar_filter_link(active:, param:, value:, count: nil, except: nil, link_class: nil, path_params: nil, title: nil)
-    content_tag :li, class: (active ? 'nav-item active' : 'nav-item'), title: title do
+    css_class = 'nav-item'
+    css_class += ' active' if active
+    css_class += ' unread-alert' if unread_alerts?(value)
```

Comment:
Perhaps here instead of making a database query we could add a css class to name each sidebar_filter_link, for example by using the `param` and `value` arguments so we get a class like `type-RepositoryVulnerabilityAlert`. 

You could change line 237 to:

    css_class += " #{param}-#{value}"